source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md

```diff
@@ -130,6 +130,213 @@ checkers:
 - Open `ElementNameMinimalLengthSniff` class in your IDE.
 - **Look for public properties**. CodeSniffer uses them to set any configuration. 
 
+That was Rule 6.
+
+## 5 more Rules
+
+### 1. Only X Level of Indentation per Method
+
+<em class="fa fa-fw fa-times"></em>
+
+```php
+foreach ($sniffGroups as $sniffGroup) {
+    foreach ($sniffGroup as $sniffKey => $sniffClass) {
+        if (! $sniffClass instanceof Sniff) {
+            throw new InvalidClassTypeException;
+        }
+    }
+}
+```
+
+<em class="fa fa-fw fa-check"></em>
+
+```php
+foreach ($sniffGroups as $sniffGroup) {
+    $this->ensureIsAllInstanceOf($sniffGefroup, Sniff::class);
+}
+
+// ...
+private function ensureIsAllInstanceOf(array $objects, string $type)
+{
+    // ...
+}
+```
+
+### 2. Do Not Use "else" Keyword
+
+<em class="fa fa-fw fa-times"></em>
+
+```php
+if ($isEnabled) {
+    return true;
+} else {
+    return false;
+}
+```
+
+<em class="fa fa-fw fa-check"></em>
+
+```php
+if ($isEnabled) {
+    return true;
+}
+
+return false;
```

Comment:
In the book they give two example.
A ternary like here finally or two different actions.

Anything like that?

```php
if ($status == self::DONE) {
    $this->finish();
    return;
}
$this->advance($step);
```